allennlp/commands/evaluate.py

```diff
@@ -0,0 +1,91 @@
+
+def add_subparser(parser: argparse._SubParsersAction) -> argparse.ArgumentParser:  # pylint: disable=protected-access
+    description = '''Evaluate the specified model + dataset'''
+    subparser = parser.add_parser(
+            'evaluate', description=description, help='Evaluate the specified model + dataset')
+    subparser.add_argument('--config_file',
+                           type=str,
+                           required=True,
+                           help='path to the configuration file that trained the model')
+    subparser.add_argument('--weights_file',
+                           type=str,
+                           required=True,
+                           help='path to the file containing the trained weights')
+    subparser.add_argument('--evaluation_data_file',
+                           type=str,
+                           required=True,
+                           help='path to the file containing the evaluation data')
+    subparser.add_argument('--cuda_device',
+                           type=int,
+                           default=-1,
+                           help='id of GPU to use (if any)')
+
+    subparser.set_defaults(func=evaluate_from_args)
+
+    return subparser
+
+def evaluate(model: Model,
+             dataset: Dataset,
+             iterator: DataIterator,
+             cuda_device: int) -> Dict[str, Any]:
+    model.eval()
+
+    generator = iterator(dataset, num_epochs=1)
+    for batch in tqdm.tqdm(generator):
+        tensor_batch = arrays_to_variables(batch, cuda_device)
+        model.forward(**tensor_batch)
+
+    return model.get_metrics()
+
+def evaluate_from_args(args: argparse.Namespace) -> Dict[str, Any]:
+    # Load parameter file
+    with open(args.config_file) as config_file:
+        config = Params(replace_none(json.loads(config_file.read())))
+
+    # Load vocabulary from file
+    serialization_prefix = config.pop('serialization_prefix')
+    vocab_dir = os.path.join(serialization_prefix, 'vocabulary')
+    vocab = Vocabulary.from_files(vocab_dir)
+
+    # Load the evaluation data
+    dataset_reader = DatasetReader.from_params(config.pop('dataset_reader'))
+    evaluation_data_path = args.evaluation_data_file
+    logger.info("Reading evaluation data from %s", evaluation_data_path)
+    dataset = dataset_reader.read(evaluation_data_path)
+    dataset.index_instances(vocab)
+
+    # Set up GPU or CPU
+    cuda_device = args.cuda_device
+
+    # Instantiate model
+    model = Model.from_params(vocab, config.pop('model'))
+    model_state = torch.load(args.weights_file,
+                             map_location=device_mapping(cuda_device))
+    model.load_state_dict(model_state)
+    model.eval()
+
+    iterator = DataIterator.from_params(config.pop("iterator"))
+
+    metrics = evaluate(model, dataset, iterator, cuda_device)
+
+    for key, metric in metrics.items():
+        logger.info("%s: %s", key, metric)
```

Comment:
You could consider a logging statement above this, like `logger.info("Finished evaluating.  Metrics:")` (or make those two separate statements, or whatever).  By default models return no metrics, so it'd be easy to get an empty output and be confused as to why.